clang/lib/Driver/Driver.cpp

```diff
@@ -826,6 +826,18 @@ static bool addSYCLDefaultTriple(Compilation &C,
 
 // Prefix for Intel GPU specific targets used for -fsycl-targets
 constexpr char IntelGPU[] = "intel_gpu_";
+constexpr char NvidiaGPU[] = "nvidia_gpu_";
+
+static llvm::Optional<StringRef> isGPUTarget(StringRef Target) {
+  if (Target.startswith(IntelGPU)) {
+    return tools::SYCL::gen::resolveGenDevice(
+        Target.drop_front(sizeof(IntelGPU) - 1));
+  } else if (Target.startswith(NvidiaGPU)) {
+    return tools::SYCL::gen::resolveGenDevice(
+        Target.drop_front(sizeof(NvidiaGPU) - 1));
+  }
```

Comment:
This will allow items like `intel_gpu_sm20` and `nvidia_gpu_pvc`, is that intentional?